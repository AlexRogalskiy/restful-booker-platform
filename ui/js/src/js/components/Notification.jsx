import React from 'react';
import { API } from '../libs/Api.js';

let style = {
    color: 'white',
    borderRadius: '100%',
    backgroundColor: 'darkred',
    padding: '3px',
    fontWeight: 'bold',
    fontFamily : 'Arial',
    position: 'absolute',
    margin: '-5px 0px 0px -12px',
    fontSize : '0.8rem'
}

export default class Notification extends React.Component {

    constructor() {
        super();

        this.state = {
            count : 0
        }
    }

    componentDidMount() {
        API.getNotificationCount(this);
    }

    render() {
        let alertIcon;

        if(this.state.count > 0){
            alertIcon = <span className="nav-link"><i className="fa fa-inbox" style={{fontSize : "1.5rem"}}><span style={style}>{this.state.count}</span></i></span>
        } else {
            alertIcon = <span className="nav-link"><i className="fa fa-inbox" style={{fontSize : "1.5rem"}}></i></span>
        }
        return alertIcon;
    }
    
}