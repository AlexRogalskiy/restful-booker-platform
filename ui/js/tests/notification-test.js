import React from 'react';
import Notification from '../src/js/components/Notification.jsx';
import nock from 'nock';
import { waitForState } from 'enzyme-async-helpers';

beforeEach(() => {
    nock('http://localhost')
        .get('/notification/count')
        .reply(200, {
                count : 34
            }
        );
});

test('Notification renders plain inbox when no unread notifications', () => {

    const notificationComponent = shallow(<Notification />);

    expect(notificationComponent).toMatchSnapshot();

});

test('Notification renders alert inbox when there are unread notifications', async () => {

    const notificationComponent = shallow(<Notification />);
    
    await waitForState(notificationComponent, state => state.count === 34);

    expect(notificationComponent).toMatchSnapshot();

});
