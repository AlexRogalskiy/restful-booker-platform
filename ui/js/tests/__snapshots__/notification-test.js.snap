// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders alert inbox when there are unread notifications 1`] = `
<span
  className="nav-link"
>
  <i
    className="fa fa-inbox"
    style={
      Object {
        "fontSize": "1.5rem",
      }
    }
  >
    <span
      style={
        Object {
          "backgroundColor": "darkred",
          "borderRadius": "100%",
          "color": "white",
          "fontFamily": "Arial",
          "fontSize": "0.8rem",
          "fontWeight": "bold",
          "margin": "-5px 0px 0px -12px",
          "padding": "3px",
          "position": "absolute",
        }
      }
    >
      34
    </span>
  </i>
</span>
`;

exports[`Notification renders plain inbox when no unread notifications 1`] = `
<span
  className="nav-link"
>
  <i
    className="fa fa-inbox"
    style={
      Object {
        "fontSize": "1.5rem",
      }
    }
  />
</span>
`;
